streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -1294,22 +1293,26 @@ sealed trait ZChannel[-Env, -InErr, -InElem, -InDone, +OutErr, +OutElem, +OutDon
   final def toPullIn(
     scope: => Scope
   )(implicit trace: Trace): ZIO[Env, Nothing, ZIO[Env, OutErr, Either[OutDone, OutElem]]] =
-    ZIO.uninterruptible {
+    ZIO.suspendSucceed {
```

Comment:
You can change the code above to `ZIO.scopeWith(self.toPullIn(_))` as well

Reply:
Done